engine/src/main/java/com/datatorrent/stram/client/StramAppLauncher.java

```diff
@@ -543,8 +543,9 @@ public URLClassLoader loadDependencies()
 
   private void setTokenRefreshCredentials(LogicalPlan dag, Configuration conf) throws IOException
   {
-    String principal = StramUserLogin.getPrincipal();
-    String keytabPath = conf.get(StramClientUtils.KEY_TAB_FILE);
+    String principal = conf.get(StramClientUtils.TOKEN_REFRESH_PRINCIPAL, StramUserLogin.getPrincipal());
+    String keytabPath = principal.equals(StramUserLogin.getPrincipal()) ? conf.get(StramClientUtils.KEY_TAB_FILE, null) : conf.get(StramClientUtils.TOKEN_REFRESH_KEYTAB, null);
```

Comment:
Isn't it what the line I specified above in the comment would do? Why do you need to check principle.equals(StramUserLogin.getPrincipal())